doc/api/packages.md

```diff
@@ -289,29 +289,45 @@ import submodule from 'es-module-package/private-module.js';
 // Throws ERR_PACKAGE_PATH_NOT_EXPORTED
 ```
 
-Entire folders can also be mapped with package exports:
+### Subpath export patterns
+
+> Stability: 1 - Experimental
+
+Explicitly listing each exports subpath entry is recommended for packages with
+a small number of exports. But for packages that have very large numbers of
+subpaths this can start to cause package.json bloat and maintenanec issues.
+
+For these use cases, subpath export patterns can be used instead:
 
 ```json
 // ./node_modules/es-module-package/package.json
 {
   "exports": {
-    "./features/": "./src/features/"
+    "./features/*": "./src/features/*.js"
   }
 }
 ```
 
-With the preceding, all modules within the `./src/features/` folder
-are exposed deeply to `import` and `require`:
+The left hand matching pattern must always end in `*`, and all instances of `*`
```

Comment:
is there going to be a first-part-code warning for authors that omit the star and just end with a slash?

Reply:
I think we can complete this process in phases. This PR now effectively includes a "quiet" docs deprecation.

The next phase would be to make an explicit deprecation with potential warnings, then finally to remove the implementation.

There are now 4464 packages on npm shipping exports so I don't think we want to rush this or start emitting warnings yet. It will be a multi-major process at this point so I want to focus on getting the baseline out first and then follow up with the phased deprecation work.